projects/common/src/navigation/navigation.service.ts

```diff
@@ -60,6 +61,53 @@ export class NavigationService {
     return this.currentParamMap.getAll(parameterName);
   }
 
+  public navigateToUrl(url: string): Observable<boolean> {
```

Comment:
I feel like adding these to our current nav is redundant - do we think it would be feasible to condense them while adding this support in a very similar way to what's currently being done?

Current list of public methods (that would be changed)
```
public navigateWithinApp(path: NavigationPath, navigationExtras?: WithinAppNavigationExtras): Observable<boolean>
public navigateExternal(url: string, navigationType: NavigationType = NavigationType.SameWindow): Observable<boolean>
public buildParamsForNavigation(preserveParameters: string[] = []): QueryParamObject;
```
proposed:
```
public navigate(url: string): Observable<boolean>; // Adding overload, would just call convertUrlToNavParams first
public navigate(params: NavigationParams): Observable<boolean>;
public convertUrlToNavParams(url: string): NavigationParams;
private buildQueryParams(preserveParameters: string[] = []): QueryParamObject; // Renamed and private

type NavigationParams = InAppNavigationParams | ExternalNavigationParams;
interface InAppNavigationParams {
  navType: 'in-app'
  path: NavigationPath;
  replaceCurrentHistory?: boolean;
  queryParams?: QueryParamObject;
  queryParamsHandling?: 'merge' | 'preserve';
  relativeTo?: ActivatedRoute;
}

interface ExternalNavigationParams {
  navType: 'external'
  url: string;
  windowHandling: ExternalNavigationWindowHandling; // Currently an enum called NavigationType
  queryParams?: QueryParamObject;
}

```

Reply:
okay looking

Reply:
So i feel like I want to take this up separately. The `navigateToUrl` is not needed for my current change but I thought I might as well add it. It was poorly designed. The purpose of current pr is only to build angular router param and use it with routerlink.

I am going to remove this method for now and will create a separate story/pr with your suggested cleanup.